Add openrollup-mvp-phase-1 application (#912)
w3f/Grants-Program: applications/openrollup-mvp-phase-1.md

- Who is your target audience (parachain/dapp/wallet/UI developers, designers, your own user base, some dapp's userbase, yourself)?
- What need(s) does your project meet?

1) The application developers that are in search of scaling their blockchain application, for example, the payment application.
2) The game developers who do not want to interact with the chain frequently only need to submit in batches for a period of time, which saves the cost of using the chain and improves the user experience.

- Are there any other projects similar to yours in the Substrate / Polkadot / Kusama ecosystem?

There are many zk-rollup projects, such as zksync, starknet, hermez, plonky2, they are basically in the Ethereum ecosystem. They are all operated by a central operator, and in our protocol, each zkapp is an operator.

## Team :busts_in_silhouette:

### Team members

- Chris Cen (Lead developer)
- MJewzk (Developer)

### Contact

- **Contact Name:** Chris Cen
- **Contact Email:** chris@icen.cc
- **Website:** https://openrollup.org/ (not yet established)

### Legal Structure

- **Registered Address:**  not yet available
- **Registered Legal Entity:** not yet available

### Team's experience

> Our team  have much experience in applied cryptography, blockchain research, and software development. 

Reviewer: ```suggestion
Our team has a lot of experience in applied cryptography, blockchain research, and software development. 
```

> Chris has been a software engineer for over many years in the gaming industry and has led the development of various multiplayer online games. Recently he develop a hybrid consensus of Progpow and AuthorityRound [https://github.com/chrisicen/openethereum-unity](https://github.com/chrisicen/openethereum-unity). He is currently devoted to the development of off-chain ZK scalability solution for blockchain.

Reviewer: ```suggestion
Chris has been a software engineer for many years in the gaming industry and has led the development of various multiplayer online games. Recently he develop a hybrid consensus of Progpow and AuthorityRound [https://github.com/chrisicen/openethereum-unity](https://github.com/chrisicen/openethereum-unity). He is currently devoted to the development of an off-chain ZK scalability solution for blockchain.
```

## Development Roadmap :nut_and_bolt:

### Overview

- **Total Estimated Duration:** 2.7 months
- **Full-Time Equivalent (FTE):** 1.5 FTE
- **Total Costs:** 48,600 USD).

### Milestone 1 — Implement Substrate Modules

- **Estimated duration:** 2.7 month
- **FTE:**  1.5
- **Costs:** 48,600 USD

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| 0a. | License | Apache 2.0 |
| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
| 0e. | Article | We will publish an **article**/workshop that explains what was done/achieved as part of the grant. |

> | 1.  | Setup | We provide a repository that forms the basis of our deliverables.|

Reviewer: I'd suggest removing this deliverable, as it doesn't describe what you'll deliver but in what form (git repo).

Reviewer: ✅

> | 2.  | Core Types | We provide the core data types, as specified in Project Details. |

Reviewer: So these are the data structures, that will be used in your pallet? If so, I think it's sufficient to integrate it into the pallet deliverable. If there is a specific reason to do it in a separate fashion, e.g. in a dedicated library, pls elaborate on the reasons why you'd want to do it this way.

Reviewer: ✅

| 3.  | Verifier trait and mock verifier | We provide the verifier trait suitable for general apps and a mock verifier that always pass the zk-verification. |

> | 4.  | Pallet functions | We provide the functionalities(zkapp registration/user deposit/user exit/user full exit/zkapp batch submit/zkapp management/zkapp info api) which support currencies/tokens/NFTs as specified in Project Details. |

Reviewer: This both sounds to me like you're describing the pallet itself. If that's the case, feel free to merge the two deliveries into 1.

Author: hi, @takahser , I would like our pallet to support various zkvms, and the recent revision added miden verifier support in this delivery,  so it's a little clearer to keep a separate delivery.
